### ISSUE TITLE
ESP in AP mode

### ISSUE THREAD

**Issue by jajen1977** [Resinchem/ESP-Parking-Assistant#16]
Why is the esp in AP open wifi mode all the time? Anyone can connect and hack this.

**Comment by Resinchem** [comment]
Do you have special characters in your SSID or password?  If so, then it is probably related to this issue #13.  However, I am not the developer or maintainer of the ESP8266 wifi library.  If this is your issue, you should post the issue to their Github repo.

I've had no issue with multiple ESP8266 type boards and multiple builds with the ESP remaining in AP mode after entering in the proper wifi credentials as part of the [first time setup steps](https://github.com/Resinchem/ESP-Parking-Assistant/wiki/03-First-Time-Setup)

**Comment by jajen1977** [comment]
No, I don't have special characters in SSID / password

**Comment by Resinchem** [comment]
I'm happy to try to help, but I need more information before I'm able to assist.  The AP hotspot should ONLY be broadcast when it cannot connect to your local wifi... that is clearly stated in the code here:

```  //fetches ssid and pass and tries to connect
  //if it does not connect it starts an access point with the specified name "ESP_ParkingAsst"
  //If not supplied, will use ESP + last 7 digits of MAC
  //and goes into a blocking loop awaiting configuration. If a password
  //is desired for the AP, add it after the AP name (e.g. autoConnect("MyApName", "12345678")
  if (!wifiManager.autoConnect("ESP_ParkingAsst")) {  //
```
If you are seeing the AP being broadcast, then that means it is not connecting to your wifi and that is a different troubleshooting process.  If that is the case, then I'd need to know what you tried, if you are seeing the MAC or IP address appear in your router at all, what router you are using, whether you've tried moving the controller closer to the router/wireless access point, etc.  

Again, the local AP/hotspot is only broadcast on initial setup and then only if it fails to connect to wifi.  If you are truly concerned about this open access point being "hackable", then you can simply add your own password to the hotspot here: ```wifiManager.autoConnect("ESP_ParkingAsst", "mypassword")```
as it is clearly stated within the code comments.  I also just verified three different installations and none of these are broadcasting the "open AP" since they are joined to my wifi.  So you need to first look at any issues with the controller joining your wifi before stating that the AP is always open... it is not if you are joined to wifi.

**Comment by jajen1977** [comment]
I am connected to wifi all the time. The signal is strong and stable. Im seeing the IP, MAC, mqtt works fine. But at the same time esp is broadcasting SSID "ESP-66E3C2" I don't understand it but it is like this all the time. Tomorrow I will try to factory reset and I will set up everything again. Maybe it will help.

**Comment by Resinchem** [comment]
Well... any possibility that you have another ESP device running somewhere?  The AP hotspot for the Parking Assistant is "ESP_ParkingAsst" and not ESP-nnnnnn.  It's not entirely impossible that it would broadcast the ESP and MAC address, but it really shouldn't.   And even it if did for some reason, that AP should go away once it has joined wifi.

Can you try this?  Join this ESP-66E2C2 wifi on your phone and open a browser and go to 192.168.4.1.  Does it say "ESP_ParkingAsst" under the WifiManager heading... and do you see an input field for UniqueDeviceName?  If you don't see these fields, then it is not the Parking Assistant ESP broadcasting the hotspot and some other ESP device.

Please let me know what you find.  I will often see an ESP-nnnnn being broadcast from some ESP I was testing with but never onboarded to my wifi.  Again, I'm not saying it is NOT the parking assistant, but I'm trying to eliminate possibilities.

**Comment by jajen1977** [comment]
There is no other esp. It's definitely the same. I know that, because when the assistant's power is turned off, this SSID disappears. Tomorrow I'll connect to ESP-66E2C2 and let you know.

**Comment by jajen1977** [comment]
I connected  to ESP-66E2C2  and after entering the address 192.168.4.1 it say "Controller Settings (parkasst)". Now im sure that is park assist esp device.

**Comment by Resinchem** [comment]
OK... thanks for checking that.  It is strange if it is truly on your wifi that it is still broadcasting the AP.  Can you try something else?  After powering on the device and seeing the AP appear, enter the IP address of the controller that your router said it has assigned (you said you are seeing the IP address and MAC address in your router, correct?).  If you enter in that IP address, it should bring up the web interface for the parking assistant (for example, mine is 192.168.1.105): 
![ParkAssist_Clip](https://user-images.githubusercontent.com/55962781/233803914-9f27677e-46c5-4ae6-8a32-b50cadb4c8c4.jpg)

If you do get this page, please scroll all the way to the bottom and let me know what it says for 'Current version'.  If you don't get the above page, then try to ping the IP address.  
![ParkAssist_ping](https://user-images.githubusercontent.com/55962781/233804061-97318977-1ff0-4788-849b-90f46697b06a.jpg)


If it does not respond to a ping, then it is not actively on your network.  The router may be showing the MAC address (and the leased IP) since it was seen at least once, but that doesn't mean it is currently joined to your wifi.  Please let me know what you find.

**Comment by Resinchem** [comment]
Oh... and also let me know what ESP8266 board you are using.  Is it a Wemos D1 Mini or some other ESP board?

**Comment by jajen1977** [comment]
IP address and the web interface for the parking assistant responds. MQTT functionality works fine too. I checked it. The only thing that is strange is active openAP mode in the same time. Current version is v0.44 (ESP8266). The device is Wemos D1 Mini.

**Comment by Resinchem** [comment]
OK... after repeated reboots of my devices, I actually did get the AP to reappear even though the controller was connected to wifi.  It's not consistent, but I have confirmed that it can occur.  This led me to search the issues from the WifiManager library that I am using and I found this issue:  

[Sometimes AP Randomly Comes back on](https://github.com/tzapu/WiFiManager/issues/1220)

This issue has been open for over 2 years... and there may be others listed, but this is the initial one that I found.  I can try to review and see if there is a way that I can force the AP to turn off after a successful connection to wifi, but with other projects and priorities, I can't make any promises as to when I might get the opportunity to develop, test and deploy any changes.

In the interim, if you are overly concerned about "hacking" of this open AP (first note that the ESP8266 wifi is a very weak signal and does not broadcast very far), you can add a password to the AP as I detailed above.

Thanks for providing the additional information that allowed me to track down the issue.

**Comment by jajen1977** [comment]
thank you for trying to solve the problem.

**Comment by tcflint** [comment]
Resinchem your projects are great.  I had a few questions on this and the battery for mailbox.
Could you DM me to discuss?
Thanks

**Comment by Resinchem** [comment]
Fixed with [release v0.45](https://github.com/Resinchem/ESP-Parking-Assistant/releases/tag/v0.45).  Local AP/hotspot no longer continues to broadcast after successfully joining wifi.
